pkg/sql/distsql_plan_stats.go

```diff
@@ -71,6 +71,22 @@ var maxTimestampAge = settings.RegisterDurationSetting(
 	5*time.Minute,
 )
 
+// Dynamically determine the number of samples to collect based on the estimated
+// number of rows in the table. This formula is based on empirical data collected
+// by running the sampler with different sample sizes on a variety of
+// table sizes and observing the proportion of heavy hitters (most frequent
+// elements) represented in the sample.
```

Comment:
nit: update the comment to mention that the returned value is between 10k and 300k